### PR TITLE
added store_onsite_density_matrices option issue atomate2 #671

### PR DESCRIPTION
**Pull request by JonathanSchmidt1**
I added a store_onsite_density_matrices option to `from_vasp_outputs` and `from_vasp_files`. As a result  onsite_density_matrices from the OUTCAR will only be stored in the future if users  specifically ask for  it.
This was added to solve https://github.com/materialsproject/atomate2/issues/671 

I have run the pytest test_calculation.py locally and it passed. I also checked that switching the option on and off in the from_vasp_outputs call in the tests adds/removes the onsite_density_matrices from the outcar.
I am bit surprised that the test passed as I would have expected it to fail when the density matrices are missing but maybe that is because the density matrices of the test cases seem to be empty anyway.

### CODE REVIEW

**Comment by utf** [comment]
Hi @munrojm, is it possible to take a look at this PR? From our side it is needed to prevent extremely large objects being stored. This field is also automatically excluded by the MP drones.

**Comment by munrojm** [comment]
Looks fine to me. Once the linting is fixed, I will merge and patch release.

File: emmet-core/emmet/core/vasp/calculation.py [hunk 1]
```diff
@@ -547,7 +551,7 @@ def from_vasp_outputs(
                 temp = str(elph_poscar.name).replace("POSCAR.T=", "").replace(".gz", "")
                 elph_structures["temperatures"].append(temp)
                 elph_structures["structures"].append(Structure.from_file(elph_poscar))
-
+
```
Comment (utf) [inline emmet-core/emmet/core/vasp/calculation.py:554]:
```suggestion

```

**Review by utf — COMMENTED** [review]
I think this is the only change needed!

File: emmet-core/emmet/core/vasp/calculation.py [hunk 2]
```diff
@@ -452,7 +453,9 @@ def from_vasp_outputs(
             Different value tune the amount of data from the ionic_steps
             stored in the Trajectory.
             If not NO, the `ionic_steps` field is left as None.
-
+        store_onsite_density_matrices
+            Whether to store the onsite density matrices from the OUTCAR.
+
```
Comment (utf) [inline emmet-core/emmet/core/vasp/calculation.py:458]:
```suggestion

```

Sorry, this one too!
